### PR TITLE
Create datasets api

### PR DESCRIPTION
**Pull request by mengyangwang**
Summary
---------
previous PR : https://github.com/facebook/ThreatExchange/pull/516
Task : T88063259, this task is to create ThreatExchange setting page, This PR is to create delete api to be used in ThreatExchange setting page

Future Actions
---------
 Create UI page to use the delete api 

Test Plan
---------

1. black hmalib
2. mypy hmalib
3. python -m py.test to pass all tests
4. make docker
5. make upload_docker 
6.  terraform apply
7. test delete api in local and verify the result in DynamoDB

https://user-images.githubusercontent.com/81996660/116912102-c2f41980-ac15-11eb-8c99-89f0afd4b5c6.mov

### CODE REVIEW

File: hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def to_json(self) -> t.Dict:
 
 
 @dataclass
-class SyncDatasetResponse(JSONifiable):
+class SyncDeleteDatasetResponse(JSONifiable):
```
Comment (schatten) [inline hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py:61]: nit: this seems like it is for an API that syncs-deletes  which is probably not what you were going for? Perhaps some other name would do better?

File: hasher-matcher-actioner/terraform/api/main.tf [hunk 2]
```diff
@@ -83,7 +84,7 @@ data "aws_iam_policy_document" "api_root" {
   }
   statement {
     effect    = "Allow"
-    actions   = ["dynamodb:GetItem", "dynamodb:Query", "dynamodb:Scan", "dynamodb:PutItem", "dynamodb:UpdateItem"]
+    actions   = ["dynamodb:GetItem", "dynamodb:Query", "dynamodb:Scan", "dynamodb:PutItem", "dynamodb:DeleteItem"]
```
Comment (schatten) [inline hasher-matcher-actioner/terraform/api/main.tf:87]: You need update AND delete, right?

**Review by schatten — APPROVED** [review]
Approve to unblock. If you have a strong reason to remove updateItem privs, go ahead.

There is a nit that you may want to address, but totally up to you.

File: hasher-matcher-actioner/terraform/api/main.tf (hunk 2 above)
Comment (mengyangwang) [reply hasher-matcher-actioner/terraform/api/main.tf:87]: actually for now we just need put_item and delete_item, update_item is not needed for now

File: hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py (hunk 1 above)
Comment (mengyangwang) [reply hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py:61]: yeah I agree! I will change this before I merge it to master

**Pull request merged**